### ISSUE TITLE
Avoiding having to define Pydantic and SQLAlchemy models separately?

### ISSUE THREAD

**Issue by Mahi** [tiangolo/fastapi#721]
Following the [SQL (Relational) Databases tutorial](https://fastapi.tiangolo.com/tutorial/sql-databases/), it seems one has to define Pydantic and SQLAlchemy models separately:
```py
# models.py
class User(Base):
    __tablename__ = 'users'
    id = Column(Integer, primary_key=True)
    name = Column(String)

# schema.py
class User(BaseModel):
    name: str
```
As the number of models grows, this leads to large amounts of duplicate code and attributes. Is there a way to merge these two classes into one, or are there any plans on adding such functionality?

**Comment by acnebs** [comment]
https://github.com/tiangolo/fastapi/issues/214

**Comment by tiangolo** [comment]
@Mahi yeah, except that SQLAlchemy models could have private data like hashed passwords. And Pydantic models wouldn't.

Also, input Pydantic models would probably have data that is not stored in the DB, like a password.

But still, this is probably a good idea for a third-party package.

It wouldn't make sense to have it as part of FastAPI, as FastAPI is independent of ORM. And it wouldn't make sense to have it as part of Pydantic for the same reason.

But it would definitely be great (at least for some use cases) to have a third-party package that is based on Pydantic and generates SQLAlchemy models from them.

---

@timothycrosley here's an example of this idea that we discussed recently :smile: And I'm pretty sure it's not the first time it pops up...
